playwright-tests/tests/treasury-factory/treasury-factory.spec.js

```diff
@@ -126,11 +126,35 @@ test("should be able to create a treasury instance with sandbox, and create/exec
     ).length,
   ).toBe(0);
 
+  const preload_content = {};
+  preload_content[`${instance_name}.near`] = 
+  {"widget":
+    {"app":
+      {"metadata":
+        {
+          "description": "test description",
+          "image":{"ipfs_cid":"bafkreido4srg4aj7l7yg2tz22nbu3ytdidjczdvottfr5ek6gqorwg6v74"},
+          "name": "test title",
+          "tags": {"devhub":"","communities":"","developer-governance":"","app":""}
+        }
+      }
+    }
+  };
+  
+  const preloads = {};
+  preloads[`/web4/contract/social.near/get?keys.json=%5B%22${instance_name}.near/widget/app/metadata/**%22%5D`] = {
+"contentType": "application/json",    
+"body":Buffer.from(JSON.stringify(preload_content)).toString('base64')
+  }; 
+
   const web4GetResult = await sandbox.account.viewFunction({
     contractId: `${instance_name}.near`,
     methodName: "web4_get",
-    args: { request: { path: "/" } },
+    args: {
+      request: { path: "/", preloads }      
+    }
```

Comment:
can you help me understand, so if we don't have this? the `web4_get` will fail?
because is [frontend](https://github.com/NEAR-DevHub/neardevhub-treasury-dashboard/blob/staging/instances/treasury-factory.near/widget/components/create-treasury/SummaryStep.jsx#L102) to make sure the web4 contract is deployed we use the same method

Reply:
Yes, preloads is a mechanism of web4 to load external data. When the web4 gateway calls web4_get it will return the URLs of the external resources. The web4 gw will then fetch from those URLs and call web4_get again with the content. As it is done in this test.